### PR TITLE
Fix golang IDs

### PR DESCRIPTION
**Pull request by enfipy**
Fix IDs for alt, vtb, vbk, btc

### CODE REVIEW

File: bindings/go/api/extern.go [hunk 1]
```diff
@@ -20,16 +21,28 @@ func SetOnGetBlockHeaderHash(fn func(toBeHashed []byte) []byte) {
 }
 
 // SetOnGetAtv ...
-func SetOnGetAtv(fn func(id []byte) []byte) {
-	ffi.OnGetAtv = fn
+func SetOnGetAtv(fn func(idBytes entities.AtvID) []byte) {
+	ffi.OnGetAtv = func(id []byte) []byte {
+		var idBytes entities.AtvID
+		copy(idBytes[:], id)
+		return fn(idBytes)
```
Comment (Warchant) [inline bindings/go/api/extern.go:28]: https://stackoverflow.com/questions/28886616/convert-array-to-slice-in-go

File: bindings/go/api/extern.go [hunk 2]
```diff
@@ -20,16 +21,28 @@ func SetOnGetBlockHeaderHash(fn func(toBeHashed []byte) []byte) {
 }
 
 // SetOnGetAtv ...
-func SetOnGetAtv(fn func(id []byte) []byte) {
-	ffi.OnGetAtv = fn
+func SetOnGetAtv(fn func(idBytes entities.AtvID) []byte) {
+	ffi.OnGetAtv = func(id []byte) []byte {
+		var idBytes entities.AtvID
+		copy(idBytes[:], id)
```
Comment (Warchant) [inline bindings/go/api/extern.go:27]: do as little copies as possible

**Review by Warchant — CHANGES_REQUESTED** [review]

File: bindings/go/api/extern.go (hunk 1 above)
Comment (enfipy) [reply bindings/go/api/extern.go:28]: https://stackoverflow.com/questions/30285680/how-to-convert-slice-to-fixed-size-array

File: bindings/go/api/extern.go (hunk 2 above)
Comment (enfipy) [reply bindings/go/api/extern.go:27]: Sadly, it's not possible here as we have different sizes for IDs. First I tried to write a similar code in `ffi/extern.go` but the whole difference was is just more boilerplate code

**Review by Warchant — APPROVED** [review]

**Pull request merged**